lib/testers/rule-tester.js

```diff
@@ -223,7 +251,21 @@ RuleTester.prototype = {
 
             validator.validate(config, "rule-tester");
 
-            return eslint.verify(code, config, filename);
+            // To cache AST.
+            eslint.reset();
+            eslint.on("Program", function(node) {
+                beforeAST = cloneDeeplyExcludesParent(node);
+
+                eslint.on("Program:exit", function(node) {
```

Comment:
I need to add this `"Program:exit"` handler after setup rule handlers to catch whether AST was not modified in the rule's `"Program:exit"` handler.

It might be better than this to add `"onStart"` / `"onEnd"` events.